tests/components/subaru/test_init.py

```diff
@@ -0,0 +1,117 @@
+"""Test Subaru component setup and updates."""
+from datetime import timedelta
+from unittest.mock import patch
+
+from subarulink import SubaruException
+
+from homeassistant.components import subaru
+from homeassistant.components.subaru.const import (
+    CONF_HARD_POLL_INTERVAL,
+    DEFAULT_HARD_POLL_INTERVAL,
+    DEFAULT_SCAN_INTERVAL,
+    DOMAIN,
+    ENTRY_CONTROLLER,
+    ENTRY_COORDINATOR,
+)
+from homeassistant.const import CONF_SCAN_INTERVAL
+from homeassistant.setup import async_setup_component
+
+from .api_responses import (
+    TEST_VIN_1_G1,
+    TEST_VIN_2_EV,
+    TEST_VIN_3_G2,
+    VEHICLE_DATA,
+    VEHICLE_STATUS_EV,
+    VEHICLE_STATUS_G2,
+)
+from .conftest import setup_subaru_integration
+
+
+async def test_setup_with_no_config(hass):
+    """Test DOMAIN is empty if there is no config."""
+    assert await async_setup_component(hass, DOMAIN, {})
+    assert DOMAIN in hass.data
+    assert hass.data[DOMAIN] == {}
+
+
+async def test_setup_ev(hass, ev_entry):
+    """Test setup with an EV vehicle."""
+    assert hass.data[DOMAIN][ev_entry.entry_id]
+
+
+async def test_setup_g2(hass):
+    """Test setup with a G2 vehcile ."""
+    entry = await setup_subaru_integration(
+        hass,
+        vehicle_list=[TEST_VIN_3_G2],
+        vehicle_data=VEHICLE_DATA[TEST_VIN_3_G2],
+        vehicle_status=VEHICLE_STATUS_G2,
+    )
+    assert hass.data[DOMAIN][entry.entry_id]
```

Comment:
Avoid accessing `hass.data` directly in the test.

Use the standard interfaces to `hass.config_entries` to verify its setup

Reply:
updated